lessons/projects/5.mdx

```diff
@@ -0,0 +1,1379 @@
+
+  const mintToken = async (e) => {
+    try {
+      setIsMinting(true);
+      setModalShow(true);
+      let mintTxn = await mint({
+        recklesslySetUnpreparedOverrides: {
+          value: ethers.utils.parseEther(e.target.value),
+        }
+      });
+      await mintTxn.wait();
+      console.log("This is the mint data", mintData);
+      refetchTokenData();
+      setIsMinting(false);
+    } catch (error) {
+      console.log("Error minting NFT", error.message);
+    }
+  };
+
+// return statement...
+```
+
+When we trigger the `mintToken` function we want to set the state to 
+`isMinting`. Based on this condition, we can inform the users that the 
+mint is in progress. We also want our modal to pop up right after this 
+so we set the value of `modalShow` to `true`.
+
+We're there at last! Now we can render our modal.
+
+```jsx
+// return statement for rendering...
+
+      {isUserConnected ?
+        <main className={styles.main}>
+          <div style={NFTFlex}>
+
+            {/* NFT Card code... */}
+
+          </div>
+          {modalShow && (
+            <div style={modal}>
+              {isMinting ?
+                <div style={modalContent}>
+                  <h2>Minting...</h2>
+                </div>
+                :
+                <div style={modalContent}>
+                  <h2>Mint Successful</h2>
+                  <div style={modalBody}>
+                    <h3>{latestNFTMinted.name}</h3>
+                    <Image src={latestNFTMinted.image} height='200px' width='200px' />
+                  </div>
+                  <div style={modalFooter}>
+                    <button style={modalButton}>
+                      <a href={`https://testnets.opensea.io/assets/goerli/${CONTRACT_ADDRESS}/${tokenData}`} target="_blank">
+                        View on OpenSea
+                      </a>
+                    </button>
+                    <button style={modalButton}>
+                      <a href={`https://goerli.etherscan.io/tx/${mintData.hash}`} target="_blank">
+                        View on Etherscan
+                      </a>
+                    </button>
+                    <button onClick={() => setModalShow(false)} style={modalButton}>Close</button>
+                  </div>
+                </div>
+              }
+            </div>
+          )}
+        </main> 
+
+// closing <div> and styling consts...
+```
+
+This may seem like a lot of code but we're going to look at it in bits:
+- We have a heading that tells us our mint is successful.
+- The body shows us the name, id and image of the NFT minted. We fetch 
+from the information stored earlier in the `latestNFTMinted` variable.
+- We have a footer that provides links to view our NFT on OpenSea and to 
+the transaction details on Etherscan. Additionally we have a **Close** 
+button that changes the state of the `modalShow` to `false` and thus 
+closes the modal modal when we click it.
+
+The final modal should look like this:
```

Comment:
Just to let you know that there aren't any colours rendering, same as  with previous screenshot at the earlier stage:
![ViewOfFirstMintAtEndOfLesson 2022-09-24 01-58-16](https://user-images.githubusercontent.com/40469149/192072044-0f7558a7-bf9e-4334-b029-bf92000f09d0.png)

The Open Sea link on the button takes me here. You can see the /undefined path at the end of the URL: https://testnets.opensea.io/assets/goerli/0x896099f50fF8eEcAc7dEd8c1AEa3D4Ed31a0F3eA/undefined
And it just gets stuck in load page  mode
 ![FirstMint 2022-09-24 01-59-36](https://user-images.githubusercontent.com/40469149/192072064-28ff11bc-897d-49c0-a81c-953471c55ea8.png)
 
 although when I use the search bar wiht the contract address, it finds my mints: https://testnets.opensea.io/collection/tiernft-awzppdckyg